src/main/java/uk/org/tombolo/importer/osm/OSMEntityHandler.java

```diff
@@ -142,11 +141,45 @@ private void handleEntity(OsmEntity entity, Geometry osmGeometry) {
 
     }
 
+    /**
+     * When dealing with GeometryCollection-s we want to be sure that we persist on the database a valid geometry.
+     * From a valid GeometryCollection we get if possible the next element that is not empty.
+     *
+     * @param geometryCollection Geometry for the subject
+     *
+     * @return Returns the chosen geometry from the collection
+     */
+    private Geometry dumpGeometryCollection(GeometryCollection geometryCollection) {
+        Geometry element = new GeometryFactory().createGeometryCollection(null);
+        int elementIndex = 0;
+        while (elementIndex < geometryCollection.getDimension()) {
+            element = geometryCollection.getGeometryN(elementIndex);
+            if (!element.isEmpty()) {
+                return element;
+            }
+            elementIndex++;
+        }
+
+        return element;
+    }
+
     private void persistEntity(OsmEntity entity, Geometry geometry, Map<String, String> tags) {
-        if (geometry == null){
-            log.warn("Could not save Subject with null geometry");
+        // If the geometry is null or not valid, it will be skipped
+        if (geometry == null || !geometry.isValid()) {
+            log.warn("Could not build {}: {} (geometry not valid): {}", entity.getClass(), entity.getId(), geometry);
             return;
         }
+
+        // Check if it's a GeometryCollection and dump it to a chosen geometry eventually
+        if (geometry instanceof GeometryCollection && !(geometry instanceof MultiPolygon)) {
+            Geometry chosenGeo = dumpGeometryCollection((GeometryCollection) geometry);
+            if (chosenGeo.isEmpty()) {
+                log.warn("Could not build {}: {} (geometry collection contains only empty geometries): {}", entity.getClass(), entity.getId(), geometry);
+                return;
+            }
+            geometry = chosenGeo;
+        }
+
         geometry.setSRID(Subject.SRID);
```

Comment:
Just wondering, do we need to check here whether Geometry is empty or is it ok to have empty geometry?

Reply:
In theory is ok, but it does not make much sense to have them as that subject wouldn't add any value. I will add a check and log a message here.